scripts/update_package_versions.py

```diff
@@ -712,54 +588,298 @@ def bump_version_in_yaml(
+            return
+
+        old_string = str(old_public_id)
+        new_string = str(new_public_id)
+        print(
+            f"Processing file {fp} for replacing {old_string} with {new_string} (is_ambiguous: {is_ambiguous})"
+        )
+
+        content = replace_in_yamls(content, old_public_id, new_public_id, type_)
+        content = replace_in_protocol_readme(
+            fp, content, old_public_id, new_public_id, type_
+        )
+        if not is_ambiguous:
+            content = content.replace(old_string, new_string)
+        else:
+            content = self._ask_to_user_and_replace_if_allowed(
+                content, old_string, new_string, type_
+            )
+
+        with fp.open(mode="w") as f:
+            f.write(content)
+
+    def _ask_to_user_and_replace_if_allowed(
+        self, content: str, old_string: str, new_string: str, type_: str
+    ) -> str:
+        """
+        Ask to user if the line should be replaced or not, If the script arguments allow that.
+
+        :param content: the content.
+        :param old_string: the old string.
+        :param new_string: the new string.
+        :param type_: the type of the package.
+        :return: the updated content.
+        """
+        if self.option_no_interactive and self.option_replace_by_default:
+            content = content.replace(old_string, new_string)
+            return content
+
+        lines = content.splitlines(keepends=True)
+        for idx, line in enumerate(lines[:]):
+            if old_string not in line:
+                continue
+
+            can_replace = _can_disambiguate_from_context(line, old_string, type_)
+            # if we managed to replace all the occurrences, then save this line and continue
+            if can_replace is not None:
+                lines[idx] = (
+                    line.replace(old_string, new_string) if can_replace else line
+                )
+                continue
+
+            # otherwise, forget the attempts and ask to the user.
+            answer = _ask_to_user(
+                lines, line, idx, old_string, type_, self.option_context
+            )
+            if answer == "y":
+                lines[idx] = line.replace(old_string, new_string)
+        return "".join(lines)
+
+
+@click.command()
+@click.option(
+    "--ask-version",
+    "-a",
+    is_flag=True,
+    help="Ask for every package version interactively",
+)
+@click.option(
+    "--update-minor",
+    "update_version",
+    flag_value="minor",
+    default=True,
+    help="Increase minor version",
+)
+@click.option(
+    "--update-patch",
+    "update_version",
+    flag_value="patch",
+    help="Increase patch version",
+)
+@click.option(
+    "--no-interactive",
+    "-n",
+    is_flag=True,
+    help="Don't ask user confirmation for replacement.",
+)
+@click.option(
+    "--context",
+    "-C",
+    type=int,
+    help="Don't ask user confirmation for replacement.",
```

Comment:
Seems wrong

Reply:
updated help string. now it relfects the purpose of the option